hack/push_updates.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+#
+# Push updates that need to be vendored in other repositories.
+
+if ! which hub 2>/dev/null 1>&2; then
+    echo "This tool requires the hub command line app." 1>&2
+    echo "Refer to https://github.com/github/hub for installation instructions." 1>&2
+    exit 1
+fi
+
+function show_help() {
+    cat <<EOF
+push_updates.sh [-h]
+push_updates.sh [-b BRANCH] [-n] [-r REPO] [-s SHA] [-t TITLE]
+
+  -b BRANCH
+    Set the release branch to be updated. Defaults to "master".
+
+  -h
+    Shows this help message.
+
+  -n
+    Dry run mode. Do not push or create a pull-request.
+
+  -r REPO
+    Update the given repository. May be repeated.
+    Defaults to updating all cluster-api-provider repositories:
+EOF
+
+    for r in $DEFAULT_REPOS; do
+        echo "      $r"
+    done
+
+cat <<EOF
+
+  -s SHA
+    The SHA from machine-api-operator repository to push.
+    Defaults to HEAD of master.
+
+  -t TITLE
+    The commit and pull-request title to use. Defaults to
+    a message that includes the SHA.
+
+Without any arguments, creates a pull request for each repository to
+update the vendored version of machine-api-operator to refer to the
+current HEAD commit on the master branch.
+
+EOF
+}
+
+DEFAULT_REPOS="
+cluster-api-provider-baremetal
+cluster-api-provider-gcp
+"
+TITLE=""
+SHA=""
+BRANCH_NAME="master"
+DRY_RUN=false
+
+while getopts "b:hnr:s:t:" OPTION; do
+    case $OPTION in
+        b)
+            BRANCH_NAME="$OPTARG"
+            ;;
+        h)
+            show_help
+            exit 0
+            ;;
+        n)
+            DRY_RUN=true
+            ;;
+        r)
+            REPOS="$REPOS $OPTARG"
+            ;;
+        s)
+            SHA="$OPTARG"
+            ;;
+        t)
+            TITLE="$OPTARG"
+            ;;
+    esac
+done
+
+REPOS=${REPOS:-${DEFAULT_REPOS}}
+TMPDIR=${TMPDIR:-/tmp}
+WORKING_DIR=$(mktemp -d $TMPDIR/mao-push-updates.XXXX)
+cd $WORKING_DIR
+echo "Working in $WORKING_DIR"
+
+if [ -z "$SHA" ]; then
+    echo "Determining SHA for machine-api-operator repository..."
+    hub clone openshift/machine-api-operator
+    (cd ./machine-api-operator && git checkout origin/$BRANCH_NAME)
```

Comment:
I wonder if this should not be done in a temporary directory, and additionally cleaned up after completion?

Reply:
It is a temporary directory, see lines 86-87.

I didn't add a trap to clean it up because @elmiko wanted it to run in a container, and I figured everything would be removed when the container is removed. Maybe that assumption isn't safe or some folks will run this outside of the container?

Reply:
i certainly don't want to see us do extra work just for the container option. that said, it should be relatively safe to leave artifacts in a temporary directory as it will get cleaned by the o/s as needed. additionally it might help for debugging to have the artifacts.

Reply:
It certainly helped while developing the script. :-)